taskcluster/worker_images_taskgraph/optimize.py

```diff
@@ -0,0 +1,69 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from collections import defaultdict
+from functools import cache
+from pathlib import Path
+
+from taskgraph.optimize.base import OptimizationStrategy, register_strategy
+from taskgraph.util.yaml import load_yaml
+
+from worker_images_taskgraph.util.fxci import get_worker_pool_images
+
+
+def is_subpath(base: Path, target: Path) -> bool:
+    try:
+        target.relative_to(base)
+        return True
+    except ValueError:
+        return False
+
+
+@register_strategy("integration-test")
+class IntegrationTestStrategy(OptimizationStrategy):
+
+    @cache
+    def _get_image_resources(self):
+        resource_map = defaultdict(set)
+        config_dir = Path("config")
+        for path in config_dir.glob("*.yaml"):
+            config = load_yaml(path)
+            image_key = "sharedimage" if "sharedimage" in config else "image"
+            image_name = config.get(image_key, {}).get("image_name")
+            if not image_name:
+                # Unknown config format
+                continue
+
+            resources = resource_map[image_name]
+            resources.add(path)
+
+            if scripts := config.get("vm", {}).get("script_paths"):
+                resources.update(scripts)
```

Comment:
Went down a bit of a rabbit hole trying to figure out how all of the `common` scripts get added. It looks like https://github.com/mozilla-platform-ops/worker-images/blob/c95f7991e0edf596539e97c7b20848608ff19e5a/gcp.pkr.hcl has a section for each GCP image that has a number of other resources that need to be added for GCP images. https://github.com/mozilla-platform-ops/worker-images/blob/main/azure.pkr.hcl similarly references some things for Windows in `build.provisioner.source`.